workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/ObjectEntryManagerSalesforceRestController.java

```diff
@@ -0,0 +1,396 @@
+
+			return _opportunityPartnerRoleProxyEndpoint;
+		}
+
+		return null;
+	}
+
+	private String _getScope(JSONObject userAccountJSONObject) {
+		boolean channel = false;
+		boolean partner = false;
+
+		String roleName;
+
+		for (Object roleBriefObject :
+				userAccountJSONObject.getJSONArray("roleBriefs")) {
+
+			JSONObject roleBriefJSONObject = new JSONObject(
+				roleBriefObject.toString());
+
+			roleName = roleBriefJSONObject.getString("name");
+
+			if (roleName.equals(Constants.ADMINISTRATOR)) {
+				return Constants.ADMINISTRATOR;
+			}
+
+			if (Constants.getChannelRoles(
+				).contains(
+					roleName
+				)) {
+
+				channel = true;
+
+				continue;
+			}
+
+			if (Constants.getPartnerRoles(
+				).contains(
+					roleName
+				)) {
+
+				partner = true;
+			}
+		}
+
+		if (channel) {
+			return Constants.CHANNEL;
+		}
+
+		JSONArray accountBriefsJSONArray = userAccountJSONObject.getJSONArray(
+			"accountBriefs");
+
+		if (partner && (accountBriefsJSONArray.length() > 0)) {
+			return Constants.PARTNER;
+		}
+
+		return Constants.UNAUTHORIZED;
+	}
+
+	private String _getScopeString(
+		String objectDefinitionExternalReferenceCode, String scope,
+		JSONObject userAccountJSONObject) {
+
+		StringBundler sb = new StringBundler();
+
+		if (scope.equals(Constants.ADMINISTRATOR)) {
+			return sb.toString();
+		}
+
+		if (scope.equals(Constants.CHANNEL)) {
+			if (objectDefinitionExternalReferenceCode.equals(
+					Constants.LEAD_PROXY_EXTERNAL_REFERENCE_CODE)) {
+
+				sb.append("partnerAccountOwnerEmail eq \'");
+				sb.append(userAccountJSONObject.getString("emailAddress"));
+				sb.append("\'");
+
+				return sb.toString();
+			}
+
+			if (objectDefinitionExternalReferenceCode.equals(
+					Constants.
+						OPPORTUNITY_PARTNER_ROLE_PROXY_EXTERNAL_REFERENCE_CODE)) {
+
+				sb.append("channelOwnerEmail eq \'");
+				sb.append(userAccountJSONObject.getString("emailAddress"));
+				sb.append("\'");
+
+				return sb.toString();
+			}
+		}
+
+		int accountBriefsCounter = 1;
+		int accountBriefsSize = userAccountJSONObject.getJSONArray(
+			"accountBriefs"
+		).length();
+
+		for (Object accountBriefObject :
+				userAccountJSONObject.getJSONArray("accountBriefs")) {
+
+			JSONObject accountBriefJSONObject = new JSONObject(
+				accountBriefObject.toString());
+
+			if (objectDefinitionExternalReferenceCode.equals(
+					Constants.LEAD_PROXY_EXTERNAL_REFERENCE_CODE)) {
+
+				sb.append("partnerAccountId eq \'");
+			}
+
+			if (objectDefinitionExternalReferenceCode.equals(
+					Constants.
+						OPPORTUNITY_PARTNER_ROLE_PROXY_EXTERNAL_REFERENCE_CODE)) {
+
+				sb.append("accountExternalReferenceCode eq \'");
+			}
+
+			sb.append(
+				accountBriefJSONObject.getString("externalReferenceCode"));
+			sb.append("\'");
+
+			if (accountBriefsCounter < accountBriefsSize) {
+				sb.append(" or ");
+			}
+
+			accountBriefsCounter++;
+		}
+
+		return sb.toString();
+	}
+
+	private String _getStringParameters(
```

Comment:
Some of the naming and readability of the code I think can be improved. I am wondering if it would be cleaner to create an array with all of our params in it, and then at the end just convert it to a string. Not fully convinced of it yet, but just thinking out loud. There may even be some Utils in liferay to help with the query string conversion which would make it even easier